shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/dto/ExecutionCandidateDTO.java

```diff
@@ -0,0 +1,116 @@
+/**
+
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.vip.dto;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class represents the DTO for {@link VipExecutionDTO} model class.
+ */
+public class ExecutionCandidateDTO {
+    private String identifier;
+    private String name;
+    private String pipelineIdentifier;
+    private Map<String, List<String>> inputParameters;
+    private List<DatasetParameterDTO> datasetParameters;
+    private Long studyIdentifier;
+    private String outputProcessing;
+    private String processingType;
+    private String refreshToken;
+    private String client = "shanoir-ng-front";
```

Comment:
Hi, does it play a role, to fix this here by default, if the same DTO is send by the Python script potentially? Is the Python script a "simulated" front-end in your current setup?

Reply:
The python script is a simulated shanoir-uploader client if I'm not wrong. But if not precised, then we are coming from the front, so shanoir-ng-front is the default.

Reply:
I agree with Michael that a default value is not really useful here. I removed it, and added @NotNull annotations on mandatory fields